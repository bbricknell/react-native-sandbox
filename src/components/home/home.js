import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View
} from 'react-native';
import { Button } from '../button/button';

export class Home extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{this.props.text}</Text>
        <Text>{this.props.data.ip}</Text>
        <Button text="Get Info" clickAction={this.props.getText} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  title: {
    fontSize: 20,
    marginBottom: 10
  }
});
