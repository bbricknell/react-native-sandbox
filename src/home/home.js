import React, { Component } from 'react';
import {
  Button,
  StyleSheet,
  Text,
  View
} from 'react-native';

export default class Home extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{this.props.test}</Text>
        <Button title="press me" onPress={this.props.getTest} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  title: {
    fontSize: 20,
    marginBottom: 10
  },
  text: {
    color: '#333333'
  }
});
