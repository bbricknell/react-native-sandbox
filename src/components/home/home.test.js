import React from 'react';
import Renderer from 'react-test-renderer';

import { Home } from './home';

describe('Home Component', () => {
  test('renders landing message correctly', () => {
    const home = Renderer.create(<Home text="hey" getText={() => {}} data={{ip: '90'}}/>).toJSON();
    expect(home).toMatchSnapshot();
  });
});
