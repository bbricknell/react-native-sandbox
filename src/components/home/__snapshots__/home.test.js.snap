// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders landing message correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "marginBottom": 10,
      }
    }
  >
    hey
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    90
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#5F9EA0",
        "borderRadius": 50,
        "borderWidth": 2.5,
        "marginVertical": 10,
        "minHeight": 50,
        "minWidth": 100,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#5F9EA0",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Get Info
    </Text>
  </View>
</View>
`;
